internal/k8s/status.go

```diff
@@ -60,19 +62,71 @@ func (m StatusMutatorFunc) Mutate(old interface{}) interface{} {
 
 // StatusUpdateHandler holds the details required to actually write an Update back to the referenced object.
 type StatusUpdateHandler struct {
-	Log             logrus.FieldLogger
-	Clients         *Clients
-	UpdateChannel   chan StatusUpdate
-	LeaderElected   chan struct{}
-	IsLeader        bool
-	Converter       *UnstructuredConverter
-	InformerFactory InformerFactory
+	Log           logrus.FieldLogger
+	Clients       *Clients
+	UpdateChannel chan StatusUpdate
+	LeaderElected chan struct{}
+	IsLeader      bool
+	Converter     *UnstructuredConverter
+}
+
+func (suh *StatusUpdateHandler) apply(upd StatusUpdate) {
+	gvk, err := suh.Clients.KindFor(upd.Resource)
+	if err != nil {
+		suh.Log.WithError(err).
+			WithField("name", upd.NamespacedName.Name).
+			WithField("namespace", upd.NamespacedName.Namespace).
+			WithField("resource", upd.Resource).
+			Error("failed to map Resource to Kind ")
+		return
+	}
+
+	obj, err := suh.Converter.scheme.New(gvk)
+	if err != nil {
+		suh.Log.WithError(err).
+			WithField("name", upd.NamespacedName.Name).
+			WithField("namespace", upd.NamespacedName.Namespace).
+			WithField("kind", gvk).
+			Error("failed to allocate template object")
+		return
+	}
+
+	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
this looks reasonable to me. Were the conflicts due to Contour writing both conditions updates and LB address updates ~concurrently?

Reply:
I wasn't able to debug it. Rebasing onto the Kubernetes 1.19 Go client broke knative with module dependencies :(